gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/DistributedRegion.java

```diff
@@ -343,6 +343,9 @@ public boolean requiresOneHopForMissingEntry(EntryEventImpl event) {
     if (event.getVersionTag() != null && event.getVersionTag().getRegionVersion() > 0) {
       return false;
     }
+    if (isTX()) {
+      return false;
+    }
     // if we're not allowed to generate a version tag we need to send it to someone who can
```

Comment:
@suranjan Please avoid using the region isTX() which will **always** do a ThreadLocal lookup (i.e. a hashmap lookup). Instead use event.getTXState() != null check which caches TXState at the first instance (so at this level it should be present in event already).